be/src/storage/rowset/segment_iterator.cpp

```diff
@@ -1409,6 +1409,26 @@ Status SegmentIterator::_do_get_next(Chunk* result, vector<rowid_t>* rowid) {
     MonotonicStopWatch sw;
     sw.start();
 
+    // only used for CACHE SELECT, do not form any chunk to save CPU time,
+    // just read file content in `_scan_range`
+    if (_opts.lake_io_opts.cache_file_only) {
+        // read every column in this segment at once, maybe optimize this later
+        for (auto& [cid, stream]: _column_files) {
+            auto vec_or = _column_iterators[cid]->get_io_range_vec(_scan_range);
+            if (!vec_or.ok()) {
+                return vec_or.status();
+            }
+            for (auto e : *vec_or) {
+                std::unique_ptr<char[]> buf(new char[e.second]);
+                RETURN_IF_ERROR(stream->read_at_fully(e.first, buf.get(), e.second));
```

Comment:
should not do this way, causes huge memory footprint depending on io range size.

Reply:
not ready for review